tests/test_xarrayfits.py

```diff
@@ -150,20 +99,133 @@ def beam_cube(tmp_path_factory):
     yield filename
 
 
-def test_name_prefix(beam_cube):
-    """Test specification of a name prefix"""
-    (xds,) = xds_from_fits(beam_cube, prefix="beam")
-    assert xds.beam0.dims == ("beam0-0", "beam0-1", "beam0-2")
+@pytest.fixture(scope="session")
+def multiple_hdu_file(tmp_path_factory):
+    ctypes = ["X", "Y", "FREQ", "STOKES"]
+
+    def make_hdu(hdu_cls, shape):
+        data = np.arange(np.prod(shape), dtype=np.float64)
+        data = data.reshape(shape)
+        header = {
+            # "SIMPLE": True,
+            # "BITPIX": -64,
+            # "NAXIS": len(data),
+            # **{f"NAXIS{data.ndim - i}": d for i, d in enumerate(data.shape)},
+            **{f"CTYPE{data.ndim - i}": ctypes[i] for i in range(data.ndim)},
+        }
+
+        return hdu_cls(data, header=fits.Header(header))
+
+    hdu1 = make_hdu(fits.PrimaryHDU, (10, 10))
+    hdu2 = make_hdu(fits.ImageHDU, (10, 20, 30))
+    hdu3 = make_hdu(fits.ImageHDU, (30, 40, 50))
+
+    filename = str(tmp_path_factory.mktemp("multihdu") / "data.fits")
+    hdu_list = fits.HDUList([hdu1, hdu2, hdu3])
+    hdu_list.writeto(filename, overwrite=True)
+
+    return filename
+
+
+@pytest.fixture(scope="session")
+def multiple_files(tmp_path_factory):
+    path = tmp_path_factory.mktemp("globbing")
+    shape = (10, 10)
+    data = np.arange(np.prod(shape), dtype=np.float64)
+    data = data.reshape(shape)
+
+    filenames = []
+
+    for i in range(3):
+        filename = str(path / f"data-{i}.fits")
+        filenames.append(filename)
+        primary_hdu = fits.PrimaryHDU(data)
+        primary_hdu.writeto(filename, overwrite=True)
+
+    return filenames
+
+
+def multiple_dataset_tester(datasets):
+    assert len(datasets) == 3
+
+    for xds in datasets:
+        expected = np.arange(np.prod(xds.hdu.shape), dtype=np.float64)
+        expected = expected.reshape(xds.hdu.shape)
+        assert_array_equal(xds.hdu.data, expected)
+
+    combined = xarray.concat(datasets, dim="hdu-0")
+    assert_array_equal(combined.hdu.data, np.concatenate([expected] * 3, axis=0))
+    assert combined.hdu.dims == ("hdu-0", "hdu-1")
+
+    combined = xarray.concat(datasets, dim="hdu-1")
+    assert_array_equal(combined.hdu.data, np.concatenate([expected] * 3, axis=1))
+    assert combined.hdu.dims == ("hdu-0", "hdu-1")
+
+    tds = [ds.expand_dims(dim="time", axis=0) for ds in datasets]
+    combined = xarray.concat(tds, dim="time")
+    assert_array_equal(combined.hdu.data, np.stack([expected] * 3, axis=0))
+    assert combined.hdu.dims == ("time", "hdu-0", "hdu-1")
+
+
+def test_list_files(multiple_files):
+    datasets = xds_from_fits(multiple_files)
+    return multiple_dataset_tester(datasets)
+
+
+def test_globbing(multiple_files):
+    path, _ = os.path.split(multiple_files[0])
+    datasets = xds_from_fits(f"{path}{os.sep}data*.fits")
+    return multiple_dataset_tester(datasets)
+
+
+def test_multiple_unnamed_hdus(multiple_hdu_file):
```

Comment:
@o-smirnov, @landmanbester These test cases illustrate the new hdu (and dimension) naming setup.